Revise release script

sf1919 opened the pull request:
**Description of work.**
Revision of the release.py script to generate new release notes and empty folders for new seperate release notes process.

**To test:**

1. Navigate to the ```../tools/release_generator``` folder in your command line
2. run ```python release.py --release 6.5.0 --milestone "Release 6.5"```
3. Check that v6.5.0 release notes have been created, including subfolders containing .gitkeep files 


Part of #33593 

*This does not require release notes* because **this is a developer script for release**


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

sf1919 committed:
Use templates to create release notes

sf1919 committed:
Creates empty release note folders and gitkeep files

sf1919 committed:
Minor edit to ReleaseChecklist

SilkeSchomann commented:
The only subfolders without .gitkeep are all the Bugfixes subfolder within Muon. Is that intended or a problem?

sf1919 committed:
Convert to PathLib and add gitkeep files

- tidied the use of paths using pathlib
- added missing call to create .gitkeep files in Muon folders

SilkeSchomann reviewed (approved):
Now all the .gitkeep files are getting created. Looks good :-)

DavidFair reviewed (approved):
LGTM, nice switch to `PathLib` from `os.path`
